Scripts/General/input_manager.gd

```diff
@@ -29,8 +29,11 @@ func get_battle_maps_list() -> Array[String]:
 	return FileSystemHelpers.list_files_in_folder(CFG.BATTLE_MAPS_PATH)
 
 
-## starts game based on game_setup_info
-func start_game():
+func start_or_set_game_in_state(world_state : SerializableWorldState, \
```

Comment:
i'd make separate functions,
duplication is low, just separate clear_players to a separate function
there is no point in having such huge if else and force_state check

Reply:
some small rearrange